### PR TITLE
Add HCAL DPG HO filter modules

### PR DESCRIPTION
**Pull request by kfjack**
PR for the newly proposed HO filter skims from HCAL DPG.

### CODE REVIEW

**Comment by fabozzi** [comment]
please test

**Comment by fabozzi** [comment]
Is it normal that tests are not yet completed?

**Comment by smuzaffar** [comment]
no idea what happened to the test, I am restarting it.

**Comment by fabozzi** [comment]
please test

**Comment by fabozzi** [comment]
+1

**Comment by fabozzi** [comment]
@davidlange6 
when this is integrated in the release, we can include the skim in a relval for testing

File: DPGAnalysis/Skims/src/JetHTJetPlusHOFilter.cc [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+  if( phiout > 2*M_PI || phiout < -2*M_PI) {
+    phiout = fmod( phiout, 2*M_PI);
+  }
+  if (phiout <= -M_PI) phiout += 2*M_PI;
+  else if (phiout >  M_PI) phiout -= 2*M_PI;
+  
+  return phiout;
+}
+
+double JetHTJetPlusHOFilter::delta2R(double eta1, double phi1, double eta2, double phi2) {
```
Comment (davidlange6) [inline DPGAnalysis/Skims/src/JetHTJetPlusHOFilter.cc]: @kfjack - please use the already existing implementation in DataFormats/Math/interface/deltaR.

File: DPGAnalysis/Skims/src/JetHTJetPlusHOFilter.cc [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+void
+JetHTJetPlusHOFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  //The following says we do not know what parameters are allowed so do no validation
+  // Please change this to state exactly what you do use, even if it is no parameters
+  edm::ParameterSetDescription desc;
+  desc.setUnknown();
+  descriptions.addDefault(desc);
+}
+
+double JetHTJetPlusHOFilter::PhiInRange(const double& phi) {
```
Comment (davidlange6) [inline DPGAnalysis/Skims/src/JetHTJetPlusHOFilter.cc]: use reduceRange from deltaPhi.h

**Comment by kfjack** [comment]
Replaced with the standard deltaR & deltaPhi functions already. Please check. Many thanks.

**Comment by fabozzi** [comment]
please test

**Comment by fabozzi** [comment]
please test

**Comment by fabozzi** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**